RELEASE_GUIDE.md

```diff
@@ -42,50 +70,134 @@ considering what effects this may have:
+git commit -m "Bumped versions for upcoming release"
+```
 
-      - The `version` field in `lib/purescript-cst/purescript-cst.cabal` (note
-        that the new version should be based on the PVP, according to what
-        changed since the previous release, and not on the actual compiler
-        version)
+5. Create the PR on GitHub
 
-      - The versions table in `lib/purescript-cst/README.md`,
+6. Push any other needed changes to the PR until it gets merged.
 
-      - The version bounds for `purescript-cst` in `purescript.cabal`
+### Make the GitHub Release
 
-- Create a release from the releases tab in GitHub and copy in the release
-  notes. This will also create a tag, which will kick off a CI build, which
-  will upload prebuilt compiler binaries to the release on GitHub when it
-  completes. (If the CI build fails, binaries can also be built locally and
-  manually uploaded to the release on GitHub)
+Once the above PR has been merged, create a GitHub release by doing the following:
+1. From the [releases](https://github.com/purescript/purescript/releases) tab in GitHub, click the "Draft a new release" button.
+2. In the "Tag version" and "Release title" fields, type in the version of the next release (e.g. "v0.14.2").
+3. In the release notes, copy and paste the version's corresponding section in the CHANGELOG.md file.
+4. Click the "Publish Release" button
 
-- Publish to Hackage:
+This will create the tag defined above, which will kick off a CI build. When it completes, the CI build will upload prebuilt compiler binaries to the release on GitHub. You can see the CI build's progres via the [Actions](https://github.com/purescript/purescript/actions) tab under a workflow whose name matches the tag you defined previously.
 
-  - change to the `lib/purescript-cst` directory and run `stack upload .`
+If the GitHub-release-triggered CI build fails, the binaries will need to be built locally and manually uploaded to the release on GitHub. In the unlikely event this occurs, coordinate with other maintainers on how to resolve the issue.
 
-  - Finally, run `stack upload .` from the repo root directory.
+### Publishing the packages to their respective registries
 
-  It's a good idea to check that the two packages (`purescript` and
-  `purescript-cst`) can be installed from Hackage at this
-  point.
+#### Hackage Uploads
 
-- After all of the prebuilt binaries are present on the GitHub releases page,
-  publish to npm: change to the `npm-package` directory and run `npm publish`.
-  It's a good idea to check that the package can be installed from npm at this
-  point.
+```bash
+cd purescript
+
+# 1. Upload the `purescript-cst` package
+pushd purescript/lib/purescript-cst
+stack upload
+popd
+
+# 2. Upload the `purescript` package`
+stack upload
+
+# 3. Verify that the two (i.e. `purescript-cst` and `purescript`) can be installed from Hackage.
+
+mkdir zHackageUpload
+pushd zHackageUpload
+cabal install purescript
+popd
+# rm -rf zHackageUpload
```

Comment:
If we're using cabal, then there's no need to do this in a separate directory, because `cabal install xyz` installs the package globally and doesn't do anything with the current working directory. But I think we should avoid adding separate instructions here anyway, because we should be using the same instructions as we provide in INSTALL.md for installing the compiler from source.


```suggestion
# 3. Verify that `purescript-cst` and `purescript` can be installed from Hackage by following the instructions in INSTALL.md for building the compiler from source.
```